### PR TITLE
PRDT-103: Payment - Public User-Initiated Cancellation and Refund

### PR DESCRIPTION
**Pull request by davidclaveau**
PRDT-103

### Ticket URL:

#103 

### Description:

**Booking Cancellation & Transaction Refund Workflows**

-   Implemented complete booking cancellation workflow with SNS/SQS event-driven architecture

    -   Created `bookings/cancel/POST` handler for user-initiated cancellations with JWT authentication
    -   Added `bookings/cancel/subscriber` for async cancellation processing via SNS events
    -   Integrated SNS topic (`booking-notifications`) and SQS queue (`booking-cancellation-queue`) for reliable event handling
-   Implemented full transaction refund system with idempotency and validation

    -   Created `transactions/refunds/POST` handler for initiating refunds with comprehensive parameter validation
    -   Added `transactions/refunds/GET` handler for retrieving refund details
    -   Implemented `transactions/refunds/subscriber` for async refund processing via SNS events
    -   Created `transactions/refunds/processor` for Worldline refund request queueing via SQS
    -   Added 3-minute time-window duplicate detection to prevent accidental duplicate refunds
    -   Implemented refund sequence tracking and partial refund support
-   Added comprehensive test coverage

    -   Created test suites for booking cancellation (POST + subscriber handlers)
    -   Created test suites for transaction refunds (POST + subscriber handlers)
    -   Added edge case testing for authentication, ownership validation, duplicate detection, and time-window logic
    -   All tests passing with proper JWT authentication mocking

**Infrastructure & Configuration**

-   Created new `booking-workflow-stack` for managing cancellation/refund infrastructure

    -   Defined SNS topics and SQS queues for event-driven processing
    -   Configured Lambda functions with appropriate IAM permissions
    -   Added stack dependencies (core-stack, reference-data-stack)
-   Enhanced transaction methods with refund-specific logic

    -   `createRefund()` - Generates Worldline refund transactions with hash-based idempotency
    -   `createAndCheckRefundHash()` - Prevents duplicate refunds within configurable time windows
    -   `findAndVerifyTransactionOwnership()` - Validates user authorization with admin override support
    -   `updateTransactionForRefund()`- Tracks refund amounts and updates transaction status
    -   `getAllRefundsByTransactionId()` and `getRefundByRefundId()` - Refund retrieval utilities
-   Added AWS SNS utility layer for publishing events to topics

**Developer Experience Improvements**

-   Fixed Jest configuration to properly map `/opt/*` modules to `src/layers/*` (was incorrectly pointing to `lib/layers/*`)
-   Uses [GoAWS](https://github.com/Admiral-Piett/goaws) to mimic SNS/SQS.
-   Added `goaws.yaml` for local SNS/SQS testing with configured queues and topic subscriptions
-   Created local testing utility (`booking-workflow-subscriber-poller.js`) for polling SQS queues during development
-   Updated existing booking and transaction handlers to support new workflow patterns

**Code Quality & Validation**

-   Improved error handling
-   Added CVE validation for dependencies
-   Implemented behaviour consistency validation for code refactoring

Also:

-   Removed deprecated `src/handlers/bookings/resources.js` and `src/handlers/transactions/resources.js` files
-   Added some support for anonymous user transactions with verification token placeholder (needs further updating in the future)

### CODE REVIEW

File: lib/public-api-stack/public-api-stack.js [hunk 1]
```diff
@@ -67,6 +72,9 @@ const defaults = {
     corsPreflightMaxAgeSeconds: 600,
     logLevel: process.env.LOG_LEVEL || 'info',
     publicApiStageName: 'api',
+    hashKey: process.env.HASH_KEY || "asdf",
+    merchantId: process.env.MERCHANT_ID || "123",
+    worldlineWebhookSecret: process.env.WORLDLINE_WEBHOOK_SECRET || "secret"
```
Comment (cameronpettit) [inline lib/public-api-stack/public-api-stack.js]: secrets should be imported via Secrets Manager (see AdminIdentityStack for a reference of how this is achieved).

Typically we want as little as possible being set by `process.env` as it means we have to inject the variables via GH Actions (yucky).
